community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java

```diff
@@ -142,6 +142,26 @@
     @Description( "Maximum number of history files for the internal log." )
     public static final Setting<Integer> store_internal_log_max_archives = setting("store.internal_log.max_archives", INTEGER, "7", min(1) );
 
+    @Description( "Configures the maximal number of transcation interval between check-points, that will trigger a " +
+                  "check-point. The database will not check-point more often than this, but might check-point less " +
+                  "often, if takes more time to perform a check-point than the configured interval. A check-point is a " +
+                  "point in the transaction logs, from which recovery would start from. Longer check-point intervals " +
+                  "typically means that recovery will take longer to complete in case of a crash. On the other hand, a " +
+                  "longer check-point interval can also reduce the I/O load that the database places on the system, as " +
+                  "each check-point implies a flushing and forcing of all the store files. The default is '100000' for a " +
+                  "check-point every 100000 transactions." )
+    public static final Setting<Integer> check_point_max_tx_interval = setting( "dbms.checkpoint.max.tx.interval", INTEGER, "100000", min(1) );
+
+    @Description( "Configures the maximal time interval between check-points, that will trigger a check-point. " +
+                  "The database will not check-point more often than this, but might check-point less often, if it " +
+                  "takes more time to perform a check-point than the configured interval. A check-point is a point in " +
+                  "the transaction logs, from which recovery would start from. Longer check-point intervals typically " +
+                  "means that recovery will take longer to complete in case of a crash. On the other hand, a longer " +
+                  "check-point interval can also reduce the I/O load that the database places on the system, as each " +
+                  "check-point implies a flushing and forcing of all the store files. The default is '5m' for a " +
+                  "check-point every 5 minutes. Other supported units are 's' for seconds, and 'ms' for milliseconds." )
+    public static final Setting<Long> check_point_max_time_interval = setting( "dbms.checkpoint.time.interval", DURATION, "5m" );
```

Comment:
Inconsistent naming of these two settings, also to me it makes more sense to name not have `interval` last. My suggestion: `dbms.checkpoint.interval.tx` and `dbms.checkpoint.interval.time`

Also the descriptions are confusing talking about e.g. "maximal time", which to me sounds like the configuration is about specifying the maximum time between two checkpoints, which means that the db can decide to introduce _more_ frequent checkpoints than that.